SourceCode/Homework06/CRTRenderer.cpp

```diff
@@ -0,0 +1,68 @@
+#include "CRTRenderer.h"
+#include <algorithm>
+#include <iostream>
+
+
+CRTDistanceScene CRTRenderer::sortScene(const CRTScene& scene) const
+{
+    CRTDistanceScene triangleDistancePair;

```

Comment:
No need to sort in advance, just track the smallest t when iterating the triangles for intersection.